pkgs/by-name/me/memos/package.nix

```diff
@@ -0,0 +1,115 @@
+{
+  fetchFromGitHub,
+  buildGoModule,
+  stdenvNoCC,
+  nix-update-script,
+  pkgs,
+  lib,
+  pnpm,
+  buf,
+  cacert,
+  grpc-gateway,
+  protoc-gen-go,
+  protoc-gen-go-grpc,
+  protoc-gen-validate,
+}:
+let
+  version = "0.24.2";
+  src = fetchFromGitHub {
+    owner = "usememos";
+    repo = "memos";
+    rev = "v${version}";
+    hash = "sha256-DWOJ6+lUTbOzMLsfTDNZfhgNomajNCnNi7U1A+tqXm4=";
+  };
+
+  protobufsGenerated = stdenvNoCC.mkDerivation {
+    name = "memos-protobuf-gen";
+    inherit src;
+
+    nativeBuildInputs = [
+      buf
+      cacert
+      grpc-gateway
+      protoc-gen-go
+      protoc-gen-go-grpc
+      protoc-gen-validate
+    ];
+
+    buildPhase = ''
```

Comment:
All overriden phases should run the hooks. e.g. add `runHook preBuild` and `runHook postBuild` here

Reply:
done